qiskit_experiments/library/randomized_benchmarking/mirror_rb_experiment.py

```diff
@@ -0,0 +1,428 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Mirror RB Experiment class.
+"""
+import warnings
+from typing import Union, Iterable, Optional, List, Sequence, Tuple
+from numbers import Integral
+import itertools
+import numpy as np
+from numpy.random import Generator, BitGenerator, SeedSequence
+
+from qiskit.circuit import QuantumCircuit, Instruction, Barrier
+from qiskit.quantum_info.operators import Pauli
+from qiskit.providers.backend import Backend
+from qiskit.providers.options import Options
+from qiskit.exceptions import QiskitError
+from qiskit.circuit.library import CXGate, CYGate, CZGate, ECRGate, SwapGate, iSwapGate
+
+from .rb_experiment import StandardRB, SequenceElementType
+from .mirror_rb_analysis import MirrorRBAnalysis
+from .clifford_utils import (
+    compute_target_bitstring,
+    inverse_1q,
+    _clifford_1q_int_to_instruction,
+)
+from .sampling_utils import (
+    BaseSampler,
+    EdgeGrabSampler,
+    SingleQubitSampler,
+    GateTypeT,
+    GateInstruction,
+)
+
+# two qubit gates that are their own inverse
+_self_adjoint_gates = [CXGate, CYGate, CZGate, ECRGate, SwapGate, iSwapGate]
+
+
+class MirrorRB(StandardRB):
```

Comment:
Good discussion point. Ideally, I prefer `MirrorRB` should be a subclass of `BaseExperiment` (not `StandardRB`). I don't want to have a base class for RB because I prefer delegation (helper functions) rather than inheritance (subclassing) in order to reuse implementation. So if we could introduce helper functions that have what is currently implemented in `_transpiled_circuits()` and `_get_basis_gates`, it would be better. Anyway, such kind of refactoring should be done later (maybe in a follow-up PR before release).

Reply:
Agreed that `MirrorRB` is different enough from `StandardRB` that it's not really worth subclassing directly. I think helper functions sound good. Let's do it in a follow-up PR as you suggested.